File: app/src/main/java/top/mrxiaom/mirai/aoki/util/FileUtil.java
package top.mrxiaom.mirai.aoki.util;

import java.io.File;

public class FileUtil {
    public static void delAllFile(File file) {
        if (!file.isDirectory()) {
            file.delete();
            return;
        }
        for (File list : file.listFiles()) {
            if (list.isDirectory()) {
                delAllFile(list);
            }
            list.delete();
        }
        file.delete();
    }
}

File: app/src/main/java/top/mrxiaom/mirai/aoki/util/FileUtil.kt
package top.mrxiaom.mirai.aoki.util

import java.io.File

fun File.delFolder(child: String) {
    delFolder(File(this, child))
}
fun delFolder(file: File) {
    if (!file.isDirectory) {
        file.delete()
        return
    }
    for (list in file.listFiles() ?: arrayOf()) {
        if (list.isDirectory) {
            delFolder(list)
        }
        list.delete()
    }
    file.delete()
}